nav2_planner/src/planner_server.cpp

```diff
@@ -270,6 +275,15 @@ PlannerServer::computePlan()
     publishPlan(result->path);
 
     action_server_->succeeded_current(result);
+
+    auto cycle_elapsed_time = (now() - start_time).seconds();
+    if (cycle_elapsed_time > 0 && 1 / cycle_elapsed_time < expected_planner_frequency_) {
```

Comment:
I think you're better off storing `1 / expected_planner_frequency_` as `max_planner_duration` so you don't do an division here that could cause NaNs (and now you can remove the `> 0` check).

Reply:
> I think you're better off storing `1 / expected_planner_frequency_` as `max_planner_duration` so you don't do an division here that could cause NaNs (and now you can remove the `> 0` check).

I've changed this to comparing durations, `max_planner_duration` defined as 1 / `expected_planner_frequency_`, where `expected_planner_frequency_` is retrieve from parameter server.